### PR TITLE
Add EFI support

### PR DESCRIPTION
**Pull request by cfergeau**
This PR adds support for EFI for linux virtual machines
https://developer.apple.com/documentation/virtualization/vzefibootloader?language=objc

This allows to boot directly from a disk image without the need for an external kernel/initrd.
This requires macOS 13.

This is done with a new --bootloader option which replaces the (now legacy)
--kernel/--initrd/--kernel-cmdline parameters.

### CODE REVIEW

File: pkg/client/cmdline.go [hunk 1]
```diff
@@ -12,21 +12,32 @@ import (
...
+// linuxBootloader determines which kernel/initrd/kernel args to use when starting
+// the virtual machine.
+type linuxBootloader struct {
 	vmlinuzPath   string
 	kernelCmdLine string
 	initrdPath    string
 }
 
+// efiBootloader determines which kernel/initrd/kernel args to use when starting
+// the virtual machine.
```
Comment (praveenkumar) [inline pkg/client/cmdline.go:29]: Looks like it is copy/paste from LinuxBootloader.

File: cmd/vfkit/main.go [hunk 2]
```diff
@@ -25,54 +25,61 @@ import (
...
 	"github.com/crc-org/vfkit/pkg/vf"
 	"github.com/docker/go-units"
 	log "github.com/sirupsen/logrus"
 )
 
-type cmdlineOptions struct {
-	vcpus     uint
-	memoryMiB uint
+func newLegacyBootloader(opts *cmdline.Options) config.Bootloader {
+	if opts.VmlinuzPath == "" && opts.KernelCmdline == "" && opts.InitrdPath == "" {
```
Comment (praveenkumar) [inline cmd/vfkit/main.go:37]: Shouldn't this with `||` instead `&&`, if anything of those options are empty string should be nil?

Comment (cfergeau) [reply cmd/vfkit/main.go:37]: Below there is `cmd.MarkFlagsRequiredTogether("kernel", "initrd", "kernel-cmdline")` so if one is set, all must be set. The behaviour was the same before this commit, they were all required (even if that's not 100% correct, kernel should be mandatory and the other optional).
I would keep the && for now.

File: pkg/client/cmdline.go (hunk 1 above)
Comment (cfergeau) [reply pkg/client/cmdline.go]: Thanks, fixed.

File: cmd/vfkit/main.go (hunk 2 above)
Comment (praveenkumar) [reply cmd/vfkit/main.go:37]: Alright make sense.

**Review by praveenkumar — APPROVED** [review]

**Pull request merged**